editor/src/editor/layout/graph.tsx

```diff
@@ -899,4 +899,32 @@ export class EditorGraph extends Component<IEditorGraphProps, IEditorGraphState>
 
 		this.refresh();
 	}
+
+	private _setParentPreservingWorldTransform(node: Node, newParent: Node | null): void {
+		if (!(node instanceof TransformNode)) {
+			// For non-transform nodes, just set the parent directly
+			node.parent = newParent;
+			return;
+		}
+
+		// Store the current world transform
+		const worldPosition = node.getAbsolutePosition();
+		const worldRotation = node.rotationQuaternion || node.rotation.toQuaternion();
+		const worldScaling = node.absoluteScaling;
+
+		// Set the new parent
+		node.parent = newParent;
+
+		// Restore the world transform
+		node.setAbsolutePosition(worldPosition);
+
+		if (node.rotationQuaternion) {
+			node.absoluteRotationQuaternion.copyFrom(worldRotation);
```

Comment:
Is `absoluteRotationQuaternion` not computed automatically by Babylon.js? For me it was kind of a readonly property